npm/init.js

```diff
@@ -16,15 +17,36 @@ function validateNpmVersion() {
     }
 }
 
+function redirectDetectDownload(starturl) {
+    if(process.env.https_proxy && process.env.https_proxy.length > 0) {
+        var mainurl = process.env.https_proxy + starturl.replace('https://', '/https/');
+        console.log(mainurl);
+        http.get(mainurl, function(res) {
+            if(res.statusCode == 302) {
+                redirectDetectDownload(res.headers.location);
+            } else if (res.statusCode == 200) {
+                writeToFile(res)
+            } else {
+                console.log('Unexpected status code during JFrog CLI download')
+            }
+        }).on('error', function (err) {console.error(err);});
+    } else {
+        https.get(starturl, function(res) {
+            if(res.statusCode == 302) {
+                redirectDetectDownload(res.headers.location);
+            } else if (res.statusCode == 200) {
+                writeToFile(res)
+            } else {
+                console.log('Unexpected status code during JFrog CLI download')
+            }
```

Comment:
The above code lines seem to be duplicated. Can we put them in a function, which we can invoke twice?